src/components/documents/AddNewDocumentButton.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import CreateButton from '@material-ui/icons/Add';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+
+import { URL } from '../../js/constants/constants';
+import store from '../../js/store/store';
+import { loadDocument, loadDocumentList } from '../../js/actions/document-actions';
+
+class AddNewDocumentButton extends Component {
+  // since we dont have authentication and to create document, document need owner
+  // we do temporary cheat here, we do extra request for users
+  // then we use first found user id as owner and author in create new document
+  // later we remove get request and replace id with logged user id
+  createNewDocument = () => {
+    axios
```

Comment:
Why don't you move this to redux?

Reply:
i don't understand advantage of moving this to redux, if this button is only way we create new document, and after doing post request i can just select created document and reload list. It looks very simple

Reply:
I't because of separation of concerns and single responsibility (https://weblogs.asp.net/arturtrosin/separation-of-concern-vs-single-responsibility-principle-soc-vs-srp). If a project gets large you will always know where your communication with remote services happens. 

Also it helps for testing - in case you would have to write a test for this component you would have to mock these axios requests and responses. But when you separate concerns you can pass just actions with raw data as payload.

An specifically for this case - `loadDocument()` and `loadDocumentList()` are unrelated actions for this component. Actually this component is visually not interested inb any response from server at all. if you would just fire an action `CreateNewDocument()` from here, in effect (thunk or redux-observable) you could fire `loadDocumentSuccess()` and `loadDocumentListSuccess()`.

About your argument "it looks simple" - as project grows you never know what additional functionality you will have to add here, so this class may grow to a very big monster.